src/library.rs

```diff
@@ -688,6 +688,26 @@ impl Type {
             _ => &[],
         }
     }
+
+    pub fn is_fundamental(&self) -> bool {
+        match *self {
+            Type::Fundamental(_) => true,
+            _ => false,
+        }
+    }
+
+    pub fn get_real_type<'a>(&'a self, env: &'a Env) -> &'a Type {
+        match *self {
+            Type::Array(t) |
+            Type::CArray(t) |
+            Type::FixedArray(t, _) |
+            Type::PtrArray(t) |
+            Type::List(t) |
+            Type::SList(t) => return env.type_(t).get_real_type(env),
+            _ => {}
```

Comment:
This function need return not only `&'a Type` but `t.ns_id` as well.

Reply:
Testing!